swift/send-message/message-types/sms/Sources/SMS/sms_message.swift

```
@@ -0,0 +1,68 @@
+    - Parameters:
+        - message: The `Message` object containing the message content and recipient.
+    - Returns: An `HTTPClientRequest` object representing the HTTP request to send the SMS.
+    */
+    private func createRequest(message: Message) async throws-> HTTPClientRequest { 
+        let targetURL: String = "https://api.twilio.com/2010-04-01/Accounts/\(accountSid)/Messages" 
+        let credentials: String = "\(accountSid):\(authToken)"
+        var request = HTTPClientRequest(url: targetURL)
+        request.method = .POST
+        request.headers.add(name: "Content-Type", value: "application/x-www-form-urlencoded")
+        request.headers.add(name:"Authorization", value: "Basic \(Data(credentials.utf8).base64EncodedString())") // credentials
+        let bodyString: String = "From=\(twilioNumber)&Body=\(message.content)&To=\(message.recipient)"
+        let bodyData = Data(bodyString.utf8)
+        request.body = .bytes(ByteBuffer(bytes: bodyData))
+        return request
+      
+    }
+
+    /* Sends an SMS message via HTTP request.
+    - Parameters:
+        - messageRequest: The `Message` object containing the message content and recipient.
+    - Returns: `nil` if the message is successfully created, else an `Error` object representing the failure reason.
+    */
+    public func sendMessage(messageRequest: Message) async -> Error?{
+        
+        do{
+            let request =  try await createRequest(message: messageRequest)
+            let response: HTTPClientResponse = try await httpClient.execute(request, timeout: .seconds(30))
+            //messege created successfully 
+            if response.status.code == 201{   
+                try await httpClient.shutdown()          
+                return nil
+            }
+            else{
+                try await httpClient.shutdown()
+                return MessengerError.validationError(error: "Error sending message: \(response.status.code)")
```

make sure you're not swallowing twilio's error if they respond with 400/500 etc